modules/canopennode/CO_driver.c

```diff
@@ -105,6 +112,9 @@ static void canopen_rx_callback(const struct device *dev, struct can_frame *fram
 			rxMsg.DLC = frame->dlc;
 			memcpy(rxMsg.data, frame->data, frame->dlc);
 			buffer->pFunct(buffer->object, &rxMsg);
+			if (rxmsg_callback) {
+				rxmsg_callback();
```

Comment:
This suffers from TOCTOU (https://en.wikipedia.org/wiki/Time-of-check_to_time-of-use), but I guess the idea is to install this callback once and never "uninstall" it by setting it to `NULL`?

Reply:
You're right, that was the idea. But I changed it anyway, it's safer and more consistent with how callbacks are handled in the CAN drivers.